IRCT-CL/src/main/java/edu/harvard/hms/dbmi/bd2k/irct/cl/util/Utilities.java

```diff
@@ -233,17 +232,31 @@ public static String extractUserFromTokenIntrospection(HttpServletRequest req, S
 		post.setHeader("Content-Type", "application/json");
 		//Authorize into the token introspection endpoint
 		post.setHeader("Authorization", "Bearer " + token_introspection_token);
-		HttpResponse response = client.execute(post);
-		if (response.getStatusLine().getStatusCode() != 200){
-			logger.error("extractUserFromTokenIntrospection() error back from token intro host server ["
-					+ token_introspection_url + "]: " + EntityUtils.toString(response.getEntity()));
-			throw new ApplicationException("Token Introspection host server return non 200 error. Please see the log");
-		}
-		JsonNode responseContent = json.readTree(response.getEntity().getContent());
-		if (!responseContent.get("active").asBoolean()){
-			throw new NotAuthorizedException("Token invalid or expired");
+		CloseableHttpResponse response = null;
+		try {
+			response = client.execute(post);
+			if (response.getStatusLine().getStatusCode() != 200){
+				logger.error("extractUserFromTokenIntrospection() error back from token intro host server ["
+						+ token_introspection_url + "]: " + EntityUtils.toString(response.getEntity()));
+				throw new ApplicationException("Token Introspection host server return non 200 error. Please see the log");
+			}
+			JsonNode responseContent = json.readTree(response.getEntity().getContent());
+			if (!responseContent.get("active").asBoolean()){
+				throw new NotAuthorizedException("Token invalid or expired");
+			}
+			return responseContent.get(userField) != null ? responseContent.get(userField).asText() : null;
+		} catch (IOException ex){
+			logger.error("extractUserFromTokenIntrospection() IOException when hitting url: " + post
+					+ " with exception msg: " + ex.getMessage());
+		} finally {
+			try {
+				if (response != null)
+					response.close();
+			} catch (IOException ex) {
+				logger.error("extractUserFromTokenIntrospection() IOExcpetion when closing http response: " + ex.getMessage());
```

Comment:
proper way is: 
logger.error("extractUserFromTokenIntrospection() IOExcpetion when closing http response:  {}", ex.getMessage());